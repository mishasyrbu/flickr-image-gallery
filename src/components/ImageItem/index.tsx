import React from 'react';
import classNames from 'classnames';

import './styles.scss';

export interface ImageItemProps {
  placeholder?: string,
  className?: string,
  src: string,
  alt?: string,
}

const ImageItem: React.FC<ImageItemProps> = ({ className, src, alt }) => {
  return (
    <img
      className={classNames('image-item', className)}
      src={src}
      alt={alt}
    />
  );
}

export default ImageItem;
