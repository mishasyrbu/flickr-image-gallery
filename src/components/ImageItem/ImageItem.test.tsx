import React from 'react';
import { create, ReactTestRenderer } from 'react-test-renderer';

import ImageItem from './';

let snapshot: ReactTestRenderer;

beforeEach(() => {
  snapshot = create(<ImageItem src="" />);
});

describe('<ImageItem />', () => {
  test('it matches the snapshot', () => {
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});
