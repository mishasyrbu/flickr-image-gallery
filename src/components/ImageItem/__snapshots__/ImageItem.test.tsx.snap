// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageItem /> it matches the snapshot 1`] = `
<img
  className="image-item"
  src=""
/>
`;
